ddl/partition.go

```diff
@@ -1911,27 +1911,157 @@ func onTruncateTablePartition(d *ddlCtx, t *meta.Meta, job *model.Job) (int64, e
 		return ver, errors.Trace(dbterror.ErrPartitionMgmtOnNonpartitioned)
 	}
 
-	newPartitions := make([]model.PartitionDefinition, 0, len(oldIDs))
-	for _, oldID := range oldIDs {
-		for i := 0; i < len(pi.Definitions); i++ {
-			def := &pi.Definitions[i]
-			if def.ID == oldID {
-				pid, err1 := t.GenGlobalID()
-				if err1 != nil {
-					return ver, errors.Trace(err1)
+	if job.State == model.JobStateRunning && job.SchemaState == model.StateNone {
```

Comment:
Should move it to
`ddl_api.go: func (d *ddl) TruncateTablePartition(ctx sessionctx.Context, ident ast.Ident, spec *ast.AlterTableSpec) error `

Reply:
And also should consider `IsRollbackable` function. Do we need to support cancel `ActionTruncateTablePartition` job?

Reply:
OK, I will check the `IsRollbackable` function and complete the logic.